components/constellation/constellation.rs

```diff
@@ -2622,9 +2620,26 @@ where
         }
     }
 
-    fn handle_register_serviceworker(&mut self, scope_things: ScopeThings, scope: ServoUrl) {
+    /// <https://w3c.github.io/ServiceWorker/#schedule-job-algorithm>
+    /// and
+    /// <https://w3c.github.io/ServiceWorker/#dfn-job-queue>
+    ///
+    /// The Job Queue is essentially the channel to a SW manager,
+    /// which are scoped per origin.
+    fn handle_schedule_serviceworker_job(&mut self, pipeline_id: PipelineId, job: Job) {
+        let origin = job.scope_url.origin();
+
+        if self
+            .check_origin_against_pipeline(&pipeline_id, &origin)
+            .is_err()
+        {
+            return warn!(
+                "Attempt to schedule a serviceworker job from an origin not matching the origin of the job."
```

Comment:
Is there a spec link for this origin check? The spec links above don't talk about origin checking.

Reply:
Yes you're right it's not in the Spec, the idea is just to check that the origin as stored in the job message matches what the constellation has stored for the pipeline from which the message originate(it's still only limited as a check and there is a note in the function used to https://github.com/servo/servo/issues/11722)